ldp/nn/lm_config.py

```diff
@@ -0,0 +1,181 @@
+from enum import StrEnum
+from logging import getLogger
+from typing import Any, Self, TypeVar, no_type_check
+
+import torch
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from transformers import (
+    AutoModelForCausalLM,
+    AutoModelForSequenceClassification,
+    AutoTokenizer,
+    BitsAndBytesConfig,
+    PreTrainedModel,
+    PreTrainedTokenizer,
+)
+
+logger = getLogger(__name__)
+TModel = TypeVar("TModel", bound=PreTrainedModel)
+
+
+class ModelMode(StrEnum):
+    training = "training"
+    inference = "inference"
+
+
+class TorchDType(StrEnum):
+    bf16 = "bfloat16"
+    fp16 = "float16"
+    fp32 = "float32"
+    auto = "auto"
```

Comment:
```suggestion
    unspecified = "unspecified"
```

It seems like `auto` is not really the right word here, it's more like `unspecified`.

And if I am wrong, can you add an inline code comment stating what `auto` means here?

Reply:
It's already documented here: https://huggingface.co/docs/transformers/v4.48.0/en/main_classes/model#transformers.PreTrainedModel.from_pretrained.torch_dtype . Per above, I don't want to re-document parameters we are forwarding